### DISCUSSION TITLE
How to use suspense with no throw error

### DISCUSSION THREAD

**Discussion by arcc1845** [TanStack/query#6095] [Q&A] [answered]
I'd like to use `<Suspense />` when it's loading, but not throw when it's an error, but it doesn't seem to be available in v5 for the following reasons.

- Can't use `suspense` flag with `useQuery()` hook.
  > The experimental suspense: boolean flag on the query hooks has been removed.
  > -- https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5
- Can't use `throwOnError` flag with `useSuspenseQuery()` hook.
  >The same as for [useQuery](https://tanstack.com/query/v5/docs/react/reference/useQuery), except for:
  > - suspense
  > - throwOnError
  > - enabled
  > - placeholderData
  >
  > -- https://tanstack.com/query/v5/docs/react/reference/useSuspenseQuery

Is there any way to do this? If not I know this is a really rare case, but I wish it were possible.

**Comment by TkDodo** [comment, marked as answer]
> Is there any way to do this?

No, this is not possible per design. When using `suspense`, you'd expect `data` to be defined after you call the hook, because you don't need to handle loading states (handled by suspense boundary) and error states (handled by error boundaries). If we take error handling away, this wouldn't hold true.
